frontend/src/metabase/query_builder/components/QueryModals/QueryModals.tsx

```diff
@@ -0,0 +1,347 @@
+import { useCallback } from "react";
+import { t } from "ttag";
+import _ from "underscore";
+
+import Modal from "metabase/components/Modal";
+import QuestionSavedModal from "metabase/components/QuestionSavedModal";
+import { AddToDashSelectDashModal } from "metabase/containers/AddToDashSelectDashModal";
+import { MoveModal } from "metabase/containers/MoveModal";
+import { SaveQuestionModal } from "metabase/containers/SaveQuestionModal";
+import { ROOT_COLLECTION } from "metabase/entities/collections/constants";
+import getInitialCollectionId from "metabase/entities/collections/getInitialCollectionId";
+import EntityCopyModal from "metabase/entities/containers/EntityCopyModal";
+import Questions from "metabase/entities/questions";
+import { useDispatch, useSelector } from "metabase/lib/redux";
+import type { UpdateQuestionOpts } from "metabase/query_builder/actions/core/updateQuestion";
+import { CreateAlertModalContent } from "metabase/query_builder/components/AlertModals";
+import { ImpossibleToCreateModelModal } from "metabase/query_builder/components/ImpossibleToCreateModelModal";
+import NewDatasetModal from "metabase/query_builder/components/NewDatasetModal";
+import { QuestionEmbedWidget } from "metabase/query_builder/components/QuestionEmbedWidget";
+import { PreviewQueryModal } from "metabase/query_builder/components/view/PreviewQueryModal";
+import { MODAL_TYPES } from "metabase/query_builder/constants";
+import { getQuestionWithParameters } from "metabase/query_builder/selectors";
+import { FilterModal } from "metabase/querying";
+import QuestionMoveToast from "metabase/questions/components/QuestionMoveToast";
+import ArchiveQuestionModal from "metabase/questions/containers/ArchiveQuestionModal";
+import EditEventModal from "metabase/timelines/questions/containers/EditEventModal";
+import MoveEventModal from "metabase/timelines/questions/containers/MoveEventModal";
+import NewEventModal from "metabase/timelines/questions/containers/NewEventModal";
+import * as Lib from "metabase-lib";
+import type Question from "metabase-lib/v1/Question";
+import type { Alert, Card, CollectionId, User } from "metabase-types/api";
+import type {
+  QueryBuilderMode,
+  QueryBuilderUIControls,
+} from "metabase-types/store";
+
+type ModalType = typeof MODAL_TYPES[keyof typeof MODAL_TYPES];
```

Comment:
This was a source of conflict because #44513 hasn't been backported, but the type is identical:
https://github.com/metabase/metabase/blob/master/frontend/src/metabase/query_builder/constants.ts#L22